### PR TITLE
Remove claims

### PR DESCRIPTION
**Pull request by swindesr**
### Overview
This PR addresses issue #49. It adds the ability for a user to toggle between two modes when on the claim meals screen. The modes are "Claim" and "Remove". When on the "Remove" mode, the select menu will display meal items that have prior claims, and allow the user to enter an amount (up to the amount of total claim so far) to be removed from the claim.

### Strategy
The strategy I used to implement this is to treat any removal claim as a negative amount. In this way, we are not updating the database to remove previously made claims, and as such there is no validation on who can remove a claim. We are simply adding a new claim that has a negative value.

This required some small additions to the regex parsing to handle negatives, and will also have the added benefit of creating a log of removal claims in the database.

### Limitations
I feel that this method is effective quickly, but maybe does not fully represent a good model of how we want to handle a removal claim. In the instance we would like to fully update the database to erase the prior claim, we would need to create a new database method (this current solution did not require this).

### Media
payme_remove_claim

### CODE REVIEW

File: apps/food_select.py [hunk 1]
```diff
@@ -156,6 +165,10 @@ def meal_chooser(self, df_saved, food_dict):
                     # if we reached how many times meal was bought, remove
                     # meal from available options
                     meals.remove(meal)
+
+                if amount_claimed > 0.0:
```
Comment (pomkos) [inline apps/food_select.py:169]: note to self: where was this defined?

File: apps/food_select.py [hunk 2]
```diff
@@ -165,27 +178,27 @@ def meal_chooser(self, df_saved, food_dict):
...
+        return meals, claimed_meals
     
     def food_formatter(self, meal_title):
         '''
         Formats meal names so it includes amount left to claim by user.
         '''
         claimed = self.claimed
-        meal, bought = meal_title.split("(")
+
+        meal, *rest = meal_title.split("(")
```
Comment (pomkos) [inline apps/food_select.py:187]: I like this

File: apps/food_select.py [hunk 3]
```diff
@@ -203,7 +216,28 @@ def info_gather(self, names, meals, df_saved, food_dict, receipt_name):
...
+            amt_order_recorded = receipt_grpd.loc[order,'amount']
+        except:
+            amt_order_recorded = 0.0
+        
+        with cola:
+            amount = round(st.number_input(
+                f"How many? (Left to remove: {amt_order_recorded})", step=1.0, max_value=amt_order_recorded, min_value=0.0
+            ),2)
+        
+        return order, -amount
```
Comment (pomkos) [inline apps/food_select.py]: Can we merge this function with the previous, maybe with a true/false or other variable?

**Review by pomkos — CHANGES_REQUESTED** [review]
Superb! Recommend merging the functions as mentioned in comments and it is ready for merging with main.

File: apps/food_select.py (hunk 3 above)
Comment (swindesr) [reply apps/food_select.py]: I have merged the common functionality for the first column, and added two smaller functions for the add/remove claim input that are called based on mode.

**Review by pomkos — APPROVED** [review]
Merging!

**Pull request merged**